sopel/tools/__init__.py

```diff
@@ -751,6 +751,46 @@ def contains(self, key):
         return self.__contains__(key)
 
 
+class SopelIdentifierMemory(SopelMemory):
+    """Special Sopel memory that stores ``Identifier`` as key.
+
+    This is a convenient subclass of :class:`SopelMemory` that always casts its
+    keys as instances of :class:`Identifier`::
+
+        >>> from sopel import tools
+        >>> memory = tools.SopelIdentifierMemory()
+        >>> memory['Exirel'] = 'king'
+        >>> list(memory.items())
+        [(Identifier('Exirel'), 'king')]
+        >>> tools.Identifier('exirel') in memory
+        True
+        >>> 'exirel' in memory
+        True
+
+    As seen in the example above, it is possible to perform various operations
+    with both ``Identifier`` and :class:`str` objects, taking advantage of the
+    case-insensitive behavior of ``Identifier``.
+
+    .. note::
+
+        Internally, it will try to do ``key = tools.Identifier(key)``, which
+        will raise an exception if it cannot instanciate the key properly::
```

Comment:
Let's go with the "common" spelling.

```suggestion
        will raise an exception if it cannot instantiate the key properly::
```